spec/fortnox/api/mappers/base_spec.rb

```diff
@@ -111,57 +117,19 @@
     end
   end
 
-  describe 'AccountNumber' do
-    include_examples 'identity mapper', :account_number do
-      let(:value) { Fortnox::API::Types::AccountNumber[1234] }
-    end
-  end
-
   describe 'CountryCode' do
-    include_examples 'identity mapper', :country_code do
-      let(:value) { Fortnox::API::Types::CountryCode['SE'] }
-    end
-  end
-
-  describe 'Currency' do
-    include_examples 'identity mapper', :currency do
-      let(:value) { Fortnox::API::Types::Currency['SEK'] }
-    end
-  end
-
-  describe 'CustomerType' do
-    include_examples 'identity mapper', :customer_type do
-      let(:value) { Fortnox::API::Types::CustomerType['PRIVATE'] }
-    end
-  end
+    subject { Fortnox::API::Registry[:countrycodestring].call('GB') }
 
-  describe 'DiscountType' do
-    include_examples 'identity mapper', :discount_type do
-      let(:value) { Fortnox::API::Types::DiscountType['PERCENT'] }
-    end
-  end
+    it { is_expected.to eq('United Kingdom') }
 
-  describe 'Email' do
-    include_examples 'identity mapper', :email do
-      let(:value) { Fortnox::API::Types::Email['email@example.com'] }
-    end
-  end
+    describe 'special cases' do
+      context 'with SE' do
+        subject { Fortnox::API::Registry[:countrycodestring].call('SE') }
 
-  describe 'HouseworkType' do
-    include_examples 'identity mapper', :housework_type do
-      let(:value) { Fortnox::API::Types::HouseworkType['CONSTRUCTION'] }
-    end
-  end
-
-  describe 'VATType' do
-    include_examples 'identity mapper', :vat_type do
-      let(:value) { Fortnox::API::Types::VATType['SEVAT'] }
+        it 'translates code to country name in Swedish' do
+          is_expected.to eq('Sverige')
+        end
+      end
```

Comment:
Missing tests for empty string, `nil` and random crap 🙂

Reply:
Ok, found them further down, but that makes me wonder what we are testing here? If you want to see if we have a mapper for the type registered you should just test that `Fortnox::API::Registry[:countrycodestring]` is callable, you should not really call it.

In this case the function is unlikely to change over time, so this way of testing would hold in this special case, but in another case this test would break if the _functionality_ of the registered object changes, even if the mapper is still correctly registered. So you are testing too much in this test.

Reply:
We need this test to see that our mapper is registered with the correct name. It was when I wrote this test that I found out that a lot of our past mappers were unused because they were registered with name that were never looked up. But you are correct, we should not call the mapper here.

Reply:
And you just found two bugs @d-Pixie! Great review!